src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2124,4 +2124,124 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
     iterations: Int = 10000): KeyTable = {
     Skat(this, variantKeys, singleKey,  weightExpr, y, x, covariates, logistic, maxSize, accuracy, iterations)
   }
+
+  def trioMatrix(pedigree: Pedigree, completeTrios: Boolean): VariantSampleMatrix[Annotation, Annotation, Annotation] = {
+    if (sSignature != TString)
+      fatal("trio_matrix requires column keys of type String")
+    requireUniqueSamples("trio_matrix")
+
+    val filteredPedigree = pedigree.filterTo(stringSampleIds.toSet)
+    val trios = if (completeTrios) filteredPedigree.completeTrios else filteredPedigree.trios
+
+    val sampleIndices = sampleIds.zipWithIndex.toMap
+
+    val momIndices = Array.fill[Int](trios.length)(-1)
+    val dadIndices = Array.fill[Int](trios.length)(-1)
+    val kidIndices = Array.fill[Int](trios.length)(-1)
+    val kidIds = new Array[String](trios.length)
+
+    val memberAnnotationType = TStruct(
+      "id" -> TString,
+      "annotations" -> saSignature
+    )
+    val newColumnAnnotationType = TStruct(
+      "proband" -> memberAnnotationType,
+      "mother" -> memberAnnotationType,
+      "father" -> memberAnnotationType
+    )
+
+    val newSampleAnnotations = new Array[Annotation](trios.length)
```

Comment:
You have ``newColumnAnnotationType`` and ``newSampleAnnotations``.  To be consistent I'd use ``newColumnAnnotations``.